tests/test_rpms.py

```diff
@@ -522,19 +514,17 @@ def test_repush_with_state_none(config_file, lookasidedir, capsys):
     assert_that(files, not_(empty()))
 
 
-@xfail(strict=True)
 def test_get_state_with_error_other_than_enoent(tempdir):
     options = MagicMock(koji=None, source=tempfile.mkstemp(dir=tempdir)[1])
     processor = BaseProcessor(options)
     processor.workdir = tempfile.mkstemp(dir=tempdir)[1]
 
     # attempting to open state file raises generic IOError
-    with patch('__builtin__.open', autospec=True, side_effect=IOError):
+    with patch('six.moves.builtins.open', autospec=True, side_effect=IOError):
```

Comment:
This doesn't make sense to me.  My understanding of what happens when you do mock.patch('foo.bar.baz') is that, on foo.bar module, 'baz' will be pointed to a mock object for the duration of the context.

So, this mock would only be used if there was code of the form:

```python
import six.moves.builtins

f = six.moves.builtins.open(...)
```

...and there isn't such code.  I don't see any reason that a bare "open()" would be directed to the mock here.

I pulled this change locally, deleted the patch and it didn't change the outcome of the test at all, so this is consistent with this patch not doing anything useful.

Can you please investigate and remove or fix this if it is indeed useless?

Reply:
I have investigated the testcase and the reason why the patch is unnecessary is that an actual IOError (not a directory) is raised during the tested classes setup. This is because tempfile is specified as the workdir. So the test incidentally does what it's supposed to, even though it clearly wasn't the intended behaviour. I have fixed the setup and patched 'open' to raise a generic IOError, at least to prevent a user confusion about the intention of the testcase.

Also, it seems that patch doesn't work the way you expected because patching 'six.moves.builtins.open' raises a generic IOError on both Python 2 and 3. Removing the patch now causes it to raise no exceptions.

Reply:
Thanks.  I had a look into that to figure out where I went wrong and I figured out that mock.patch *does* work the way I expected, but my conclusion that it wouldn't affect a bare `open()` was wrong.  I learned the reasons why that works:

1. 'six.moves.builtins' is the same object as 'builtins', not a wrapper module as I'd thought it might be.  So this patch on six.moves.builtins, in python3, is the same as patching on builtins.

```
>>> id(builtins)
140273681132816
>>> id(six.moves.builtins)
140273681132816
```

2. builtins module is special, apparently anything set there can be looked up implicitly without having to be imported or having to use a qualified name.

```
>>> setattr(builtins, 'quux', lambda: 123)
>>> quux()
123
```